src/charm.py

```diff
@@ -44,33 +124,73 @@ def __init__(self, *args):
             ],
         )
 
+        # Dashboard related config (Grafana)
         self.dashboard_provider = GrafanaDashboardProvider(
             charm=self,
             relation_name="grafana-dashboard",
         )
 
```

Comment:
Doesn't actually matter, but this feels more like a _private attribute compared to others above

Reply:
How is it different from `prometheus_provider`?
I will leave it as is for this. We can address it when fixing Istio relation.